CommandLine/TokenType.cs

```diff
@@ -7,6 +7,8 @@ public enum TokenType
     {
         Argument,
         Command,
-        Option
+        Option,
+        DoubleDash,
```

Comment:
This generally looks great. My only suggestion is that we come up with a name that describes the meaning of the token rather than describing the characters themselves. 

According to the [POSIX utility syntax guidelines](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html):

> Guideline 10:
> The first -- argument that is not an option-argument should be accepted as a delimiter indicating the end of options. Any following arguments should be treated as operands, even if they begin with the '-' character.

So something like `EndOfArguments` or `OperandsPrefix` or something? Other suggestions?

Reply:
Thanks. I think `EndOfArguments` works the better of the two in the context of the other names.

Reply:
Sorry for the confusion. I meant to propose `Operand` as a replacement for `Unparsed`.

Reply:
I see. That makes sense. I wonder though if we should keep the terminology consistent with the `UnparsedTokens` property. I think "unparsed tokens" may be a bit less ambiguous than "operands." 

I also don't think `dotnet` necessarily uses these extra tokens as operands. They're treated as forwarded arguments and whether or not they're technically "operands" is an implementation detail of the downstream command.

Reply:
That was originally why I named it `Unparsed`, but, pedantically speaking, it is weird for a lexing operation to determine what is "parsed" :)

Naming is hard, which is why I don't get too attached to names...provided they have a modicum of sense.  Happy to go with either name or another suggestion entirely.

Reply:
The fact that it's forwarded is a `dotnet` decision, not a concern of the parser. You just convinced me that `Operand` is the best term in context. 

(Absolutely agreed that it's weird for a lexing operation to determine  what's parsed but `Lex` is internal and this enum is public, hence the desire to get it right. Maybe `Lex` should also be renamed. 😬)

Reply:
Ok I'll go with that terminology then.  I'll amend the commit and push it up when I get a break (I'm on grand jury duty all month). 😢

Reply:
Changes pushed.